mbs_results/utils.py

```diff
@@ -42,6 +47,28 @@ def read_colon_separated_file(filepath: str, column_names: List[str]) -> pd.Data
         buffer = BytesIO(file.read())
         df = pd.read_csv(buffer, sep=r"\s*:\s*", names=column_names, engine="python")
         date_string = re.findall(r"_(\d{6})", filepath)
-        df["period"] = int(date_string[0])
+        df[period] = int(date_string[0])
 
     return df
+
+
+def get_qv_df(filepath, qv_pattern="qv*.csv"):
+    """
+    Get required variables from qv data files
+
+    Parameters
+    ----------
+    filepath : str
+        location of cp and qv data files
+    qv_pattern : str
+        pattern for qv filenames
+        defaults to "qv*.csv"
+    """
+    filenames = [
+        filename for filename in listdir(filepath) if isfile(join(filepath, filename))
+    ]
+    qv_filenames = fnmatch.filter(filenames, qv_pattern)
+    df_list = [pd.read_csv(filepath + filename) for filename in qv_filenames]
+    df = pd.concat(df_list, ignore_index=True)
+
+    return df[["period", "reference", "question_no", "adjusted_value"]]
```

Comment:
Can we have the columns selection as an argument ?  This way the function would be more reproducible and can read any pattern of csvs

Reply:
I don't think we need this anymore to run the estimation as we output the estimation dataframe and merge that onto the main data. I think we should just delete this function